File: first assignment/c#/FirstAssignment/FirstAssignment/Edge.cs
namespace FirstAssignment;

public class Edge : IEquatable<Edge>, IComparable<Edge>
{
    public Vertex U { get; set; }
    public Vertex V { get; set; }
    public int Weigth { get; set; }
    public string Label { get; set; }

    public Edge(Vertex u, Vertex v, int weigth)
    {
        U = u;
        V = v;
        Weigth = weigth;
        Label = string.Empty;
    }

    public bool Equals(Edge? other)
    {
        return other != null && U == other.U && V == other.V && Weigth == other.Weigth;
    }

    public int CompareTo(Edge? other)
    {
        // If other is not a valid object reference, this instance is greater.
        if (other == null)
            return 1;
        return Weigth.CompareTo(other.Weigth);
    }

    public Vertex GetOpposite(Vertex v) => v.Equals(U) ? V : U;
}

File: first assignment/python/edge.py
from vertex import Vertex

class Edge:
    def __init__(self, u: Vertex, v: Vertex, weight: int) -> None:
        self.u = u
        self.v = v
        self.weight = weight
        self.label = ''

    def __eq__(self, other: object) -> bool:
        return isinstance(other, self.__class__) and \
            self.u == other.u and self.v == other.v and \
            self.weight and other.weight

    def get_opposite(self, v: Vertex):
        if v == self.u:
            return self.v
        elif v == self.v:
            return self.u
        raise Exception("NOT FOUND")